addon/models/learner-group.js

```diff
@@ -192,44 +265,83 @@ export default class LearnerGroup extends Model {
     if (this.isTopLevelGroup) {
       return this.title.replace(/\s/g, '');
     }
-    if (!this.allParentTitles) {
+
+    if (!this._parentData.isResolved || !this._grandparentsData?.isResolved) {
       return undefined;
     }
-    return [...this.allParentTitles, this.title].join('').replace(/\s/g, '');
+
+    return [
+      ...mapBy([...this._grandparentsData.value.reverse(), this._parentData.value], 'title'),
+      this.title,
+    ]
+      .join('')
+      .replace(/\s/g, '');
   }
 
   /**
    * A text string comprised of all learner-group titles in this group's tree.
    * This includes that titles of all of its ancestors, all its descendants and this group's title itself.
    */
   get filterTitle() {
-    if (!this.allParents || !this.allDescendants) {
+    if (
+      !this._parentData.isResolved ||
+      !this._grandparentsData?.isResolved ||
+      !this._childrenData.isResolved ||
+      !this._grandChildrenData?.isResolved
+    ) {
       return '';
     }
 
-    return [
-      ...mapBy(this.allDescendants, 'title'),
-      ...mapBy(this.allParents, 'title'),
-      this.title,
-    ].join('');
+    const up = this.isTopLevelGroup
+      ? []
+      : [this._parentData.value, ...this._grandparentsData.value];
+    const down = [...this._childrenData.value, ...this._grandChildrenData.value.flat()];
+
+    return [...mapBy(down, 'title'), ...mapBy(up, 'title'), this.title].join('');
   }
 
-  get allParents() {
-    if (this.isTopLevelGroup) {
+  async _getAllParents() {
+    const parent = await this.parent;
+    if (!parent) {
       return [];
     }
-    if (!this._parent?.allParents) {
-      return undefined;
+    const grandparents = await parent.getAllParents();
+
+    return [parent, ...grandparents];
+  }
+
+  @cached
+  get _grandparentsData() {
```

Comment:
suggesting name change to `_allAncestorsData()`.

Reply:
✅